### ISSUE TITLE
Failure to create a downloader crashes Kiwix

### ISSUE THREAD

**Issue by niemand-deu** [kiwix/kiwix-desktop#334]
If _createDownloader_ is unsuccessful (e.g. aria2 isn't installed or lacks support for the BitTorrent protocol), it doesn't fail gracefully and causes a segmentation fault. This can be replicated by running kiwix-desktop with and without aria2c on the same system.
Reworking the function might fix https://github.com/kiwix/kiwix-desktop/issues/268.

**Comment by kelson42** [comment]
@niemand-deu Aria2c is a dependence of Kiwix Desktop. Like you have discovered, Kiwix does not work properly if aria2c is not available. This is a feature and not a bug. The question is more: why do you think this is a problem?

**Comment by niemand-deu** [comment]
Completely omitting aria2 was just an example for easy replication. As mentioned previously, aria2c might be installed, but not always fully support the expected feature set: 
```
aria2c: unrecognized option '--dht-entry-point=router.bittorrent.com:6881'
Usage: aria2c [OPTIONS] [URI | MAGNET | TORRENT_FILE | METALINK_FILE]...
See 'aria2c -h'.
 curl_easy_perform() failed.

libcurl: (7) Failed to connect to localhost port 42042: Connection refused
```
Additionally there used to be an error window (https://github.com/kiwix/kiwix-desktop/issues/268#issue-499929716), which is completely missing now. If someone had enough foresight to launch kiwix-desktop over the terminal, they'd see `Cannot create downloader` blip up before the crash, but otherwise there's no visible indication for a user that something went wrong.

**Comment by kelson42** [comment]
@niemand-deu Which exact version of Kiwix do you use? Have you compiled it yourself or do you use a compiled version we distribute ourself? I can not reproduce this problem with officials packages of the software. But you are right that the handling of this error still does not work properly. I have #272.

**Comment by niemand-deu** [comment]
aria2 1.34.0-r2 with kiwix-desktop, kiwix-lib and libzim directly compiled from git.
Gentoo can compile aria2c with support for dht-entry-point if the user adds `bittorrent` to the use flags, but it isn't part of the default settings and easily missed.
On other distributions the feature would depend on the build environment having libgmp and libnettle installed.

**Comment by kelson42** [comment]
@niemand-deu I understand better now. Thank you for the explanation. But what you describe here is a Gentoo packaging issue. We don't maintain this package. This is not something for us. The packages we provide include proper version of aria2c. There is nothing more we can do here IMO.

**Comment by soloturn** [comment]
it would be nice if it would not crash imo and somehow tell the user what is wrong.

**Comment by kelson42** [comment]
@mgautierfr Duplicate of #268 ?

**Comment by kelson42** [comment]
I’m not exactly sure how it should, but would be better if it does not crash, I agree.

**Comment by kelson42** [comment]
@jetownfeve21 We need to check that again... your fix should report an error propely now.

**Comment by jetownfeve21** [comment]
This should be fixed with #351
